src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs

```diff
@@ -100,70 +94,76 @@ internal sealed class CompilationContext
                 _displayClassVariables.Values.Any(v => v.Kind == DisplayClassVariableKind.This));
         }
 
-        internal CommonPEModuleBuilder CompileExpressions(
+        internal bool TryCompileExpressions(
             ImmutableArray<CSharpSyntaxNode> syntaxNodes,
             string typeNameBase,
             string methodName,
-            DiagnosticBag diagnostics)
+            DiagnosticBag diagnostics,
+            [NotNullWhen(true)]out CommonPEModuleBuilder? module)
         {
             // Create a separate synthesized type for each evaluation method.
             // (Necessary for VB in particular since the EENamedTypeSymbol.Locations
             // is tied to the expression syntax in VB.)
             var synthesizedTypes = syntaxNodes.SelectAsArray(
-                (syntax, i, arg) => (NamedTypeSymbol)CreateSynthesizedType(syntax, typeNameBase + i, methodName, ImmutableArray<Alias>.Empty),
-                (object)null);
+                (syntax, i, _) => (NamedTypeSymbol)CreateSynthesizedType(syntax, typeNameBase + i, methodName, ImmutableArray<Alias>.Empty),
+                arg: 0);
```

Comment:
>arg: 0 [](start = 16, length = 6)

Does passing an `int` rather than a reference type cause the runtime to generate more specialization of the generic method? Consider reverting this.

Reply:
Good point.